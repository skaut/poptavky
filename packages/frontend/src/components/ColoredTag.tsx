import { css } from "@emotion/react";
import styled from "@emotion/styled";
import stringToColor from "string-to-color";

export const ColoredTag = styled("span")(
  (props: { children: string; isLight?: boolean }) => css`
    background-color: ${stringToColor(props.children)}33;
    color: ${props.isLight === true ? "white" : "#444;"};
    font-family: themix;
    font-size: 0.8rem;
    margin-right: 8px;
    padding: 0.2rem 0.4rem;
    border-radius: 4px;
    ${props.isLight === true ? "border: 1px solid #ffffff55" : ""};
  `,
);
