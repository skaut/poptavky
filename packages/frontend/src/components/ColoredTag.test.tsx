import renderer from "react-test-renderer"

import { ColoredTag } from "./ColoredTag"

describe("ColoredTag component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<ColoredTag>Tag</ColoredTag>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
