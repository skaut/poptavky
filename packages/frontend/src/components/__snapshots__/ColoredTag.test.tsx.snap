// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColoredTag component should renders correctly 1`] = `
.emotion-0 {
  background-color: #13791433;
  color: #444;
  font-family: themix;
  font-size: 0.8rem;
  margin-right: 8px;
  padding: 0.2rem 0.4rem;
  border-radius: 4px;
}

<span
  className="emotion-0"
>
  Tag
</span>
`;
